govcd/catalog.go

```diff
@@ -254,7 +254,7 @@ func (cat *Catalog) UploadOvf(ovaFileName, itemName, description string, uploadP
 	var task Task
 	for _, item := range vappTemplate.Tasks.Task {
 		task, err = createTaskForVcdImport(cat.client, item.HREF)
-		if err != nil {
+		if err != nil || task.Task.Status == "error" {
```

Comment:
If the error is nil but the task reports an error, we are hiding the error message from the user.
There should be a separate check for the task status and return accordingly.

Reply:
Good spot - tx, fixed.